pkg/resources/user_password_policy_attachment.go

```diff
@@ -0,0 +1,169 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var userPasswordPolicyAttachmentSchema = map[string]*schema.Schema{
+	"user_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "User name of the user you want to attach the password policy to",
+	},
+	"password_policy_database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Database name where the password policy is stored",
+	},
+	"password_policy_schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Schema name where the password policy is stored",
+	},
+	"password_policy_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Non-qualified name of the password policy",
+	},
+}
+
+func UserPasswordPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "Specifies the password policy to use for the current account. To set the password policy of a different account, use a provider alias.",
```

Comment:
This description was copy-pasted from account password policy resource, please update it to describe that it is user.